packages/bpk-component-modal/src/BpkModalDialog.js

```diff
@@ -36,6 +36,7 @@ export type Props = {
   wide: boolean,
   isIphone: boolean,
   showHeader: boolean,
+  flareComponent: ?Element<any>,
```

Comment:
New undocumented prop as its only to be used by the Dialog

Reply:
Should we give this a more generic name? From `BpkModalDialog`'s point of view, this is 'any arbitrary component to display at the top', not necessarily a flare. What about `headerComponent` or something like that?